creation/create_cvmfsexec_distros.sh

```diff
@@ -3,123 +3,166 @@
+			echo "Found newer version of cvmfsexec..."
+			echo "Rebuilding cvmfsexec distribution(s) using the latest version ${latest_ver}..."
+		fi
+
+		# build the distributions for cvmfsexec based on the source, os and platform combination
+		check_directory_exists "$cvmfsexec_distros"
+
+		cvmfs_configurations=($(echo "$cvmfs_configurations_list" | tr "," "\n"))
+		supported_machine_types=($(echo "$supported_machine_types_list" | tr "," "\n"))
+
+		local successful_builds=0
+		for cvmfs_src in "${cvmfs_configurations[@]}"
+		do
+			for mach_type in "${supported_machine_types[@]}"
+			do
+				echo -n "Making $cvmfs_src distribution for $mach_type machine..."
+				os=${mach_type%-*}
+				arch=${mach_type#*-}
+				if "$cvmfsexec_latest"/makedist -m "$mach_type" "$cvmfs_src" &> /dev/null ; then
+					"$cvmfsexec_latest"/makedist -o "$cvmfsexec_distros"/cvmfsexec-"${cvmfs_src}"-"${os}"-"${arch}" &> /dev/null
+					if [[ -e "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} ]]; then
+						echo " Success"
+						if tar -cvzf "$cvmfsexec_tarballs"/cvmfsexec_"${cvmfs_src}"_"${os}"_"${arch}".tar.gz -C "$cvmfsexec_distros" cvmfsexec-"${cvmfs_src}"-"${os}"-"${arch}" &> /dev/null; then
+							((successful_builds+=1))
+						fi
+					fi
+				else
+					echo " Failed! REASON: $cvmfs_src may not yet have a $mach_type build."
+				fi
+
+				# delete the dist directory within cvmfsexec to download the cvmfs configuration
+				# and repositories for another machine type
+				rm -rf "$cvmfsexec_latest"/dist
+			done
+		done
+
+		# remove the distros and latest folder under cvmfsexec.tmp
+		rm -rf "$cvmfsexec_distros"
+		rm -rf "$cvmfsexec_latest"
+	fi
+
+
+	# TODO: store/update version information in the $cvmfsexec_tarballs location for future reconfig/upgrade
+	if [[ "$successful_builds" -gt 0 ]]; then
+		# update only if there was at least one successful build of cvmfsexec
+		echo "$latest_ver" > "$cvmfsexec_tarballs"/.cvmfsexec_version
+	fi
+
+	end=$(date +%s)
+
+	runtime=$((end-start))
+	echo "Took $runtime seconds to create $successful_builds cvmfsexec distribution(s)"
+}
 
 
-# TODO: store/update version information in the $cvmfsexec_tarballs location for future reconfig/upgrade
-echo "$latest_ver" > "$cvmfsexec_tarballs"/.cvmfsexec_version
+####################### MAIN SCRIPT STARTS FROM HERE #######################
 
-end=`date +%s`
+if [[ $# -eq 0 ]]; then
+	echo "Building/Rebuilding of cvmfsexec distributions disabled!"
+	exit 0
+else
+	if [[ $1 == "--work-dir" ]]; then
+		# if no value is passed to/after the --work-dir option, return usage documentation and exit
+		[[ $# -eq 0 || $# -ne 4 ]] && usage || true
+		# otherwise, continue parsing the arguments
+		work_dir="$2"
+		shift 2
+	else
+		# if --work-dir is not passed, assume default work-dir (RPM install)
+    	work_dir="$DEFAULT_WORK_DIR"
+	fi
+	[[ $# -eq 0 || $# -eq 1 ]] && usage || true
+	configurations=$1
+	machine_types=$2
+fi
```

Comment:
Talking from purely a self-invocation of the script, if only the option is used, i.e. `$0 --work-dir DIR`, it is an illegal invocation because both `SOURCES_LIST` and `PLATFORMS_LIST` are required arguments (as can be seen in the usage documentation). Even though you did mention that I need to also keep in mind that the script should be functional outside of the usual reconfig/upgrade process, I think I ended up assuming that the script is rarely going to be invoked on its own and would mostly be invoked as part of factory reconfig/upgrade.

In our previous discussion, we did decide that configurations will be required and machine types will be optional. So I agree that it is better to accept the option-only case as OK. I'll modify this part based on your suggestion. 

One question by the way. Is it possible that when an option is passed, there are chances that the value corresponding to the option might not be passed. Asking this to make sure if this is a possible scenario when a shell script is invoked. I've handled this scenario in my implementation but wanted to know if this is really needed/necessary.

Reply:
Yes, it is possible. If you use getopts it will do some checks for you, otherwise, you have to do them as you did checking that is a directory.

Reply:
Thanks for the clarification. Given the possibility that could happen, I've retained the logic that handles such a scenario.